### PR TITLE
Issue #169 motor enable polarity

### PR DESCRIPTION
**Pull request by giseburt**
This is a PR for the fix to issue #169

### CODE REVIEW

File: g2core/stepper.cpp [hunk 1]
```diff
@@ -936,6 +936,31 @@ stat_t st_set_su(nvObj_t *nv)			// motor steps per unit (direct)
...
+    if (nv->value < IO_ACTIVE_LOW) { return (STAT_INPUT_LESS_THAN_MIN_VALUE); }
+    if (nv->value > IO_ACTIVE_HIGH) { return (STAT_INPUT_EXCEEDS_MAX_VALUE); }
+
+    uint8_t motor = _get_motor(nv->index);
+    if (motor > MOTORS) { return STAT_INPUT_VALUE_RANGE_ERROR; };
+
+    nv->value = (float)Motors[motor]->getMotorPolarity();
+    nv->valuetype = TYPE_INT;
+    return (STAT_OK);
+}
```
Comment (benlaurie) [inline g2core/stepper.cpp]: These functions do not appear to be used!

Comment (giseburt) [reply g2core/stepper.cpp]: They are used in the config_app “big table” used by the JSON and text commands.

I suspect you figured that out however due to your pull request for step polarity.

In light of that I should probably rename a few things from “motor polarity” to “enable polarity” in this pull request.

Comment (giseburt) [reply g2core/stepper.cpp]: To explain further, there are two times these get set when they’re not called explicitly via a JSON command:

In the constructor of `StepDirStepper` when the pin is setup at machine startup, then again (redundantly) when the JSON system initializes.The JSON system calls all of the table entries (that have the flags set to request it) with the default values.

File: g2core/stepper.h [hunk 2]
```diff
@@ -430,15 +431,17 @@ extern stPrepSingleton_t st_pre;            // only used by config_app diagnosti
...
     Timeout _motor_disable_timeout;         // this is the timeout object that will let us know when time is u
     uint32_t _motor_disable_timeout_ms;     // the number of ms that the timeout is reset to
+    ioMode _motor_enable_polarity;         // 0=active HIGH, 1=active LOW
     stPowerState _power_state;              // state machine for managing motor power
     stPowerMode _power_mode;                // See stPowerMode for values
 
     /* stepper default values */
-
+public:
     // sets default pwm freq for all motor vrefs (commented line below also sets HiZ)
```
Comment (benlaurie) [inline g2core/stepper.h]: Presumably this comment should also go, since `frequency` is deleted below?

File: g2core/stepper.cpp (hunk 1 above)
Comment (benlaurie) [reply g2core/stepper.cpp]: Silly me ... I didn't notice config_app diff was not expanded, so I couldn't find the calls.

Comment (benlaurie) [reply g2core/stepper.cpp]: Hmm. Does this mean I should do the step polarity similarly, rather than storing it in the config as I have done?

**Comment by benlaurie** [comment]
BTW, is there some way to talk to you guys other than in PRs? Mailing list? Slack? IM?

File: g2core/stepper.h (hunk 2 above)
Comment (giseburt) [reply g2core/stepper.h]: Good catch, yes, this should go as well.

**Comment by giseburt** [comment]
@benlaurie You can contact me via gmail. Same as my username before the @ sign. I'm also often on hangouts, same account.

File: g2core/stepper.cpp (hunk 1 above)
Comment (giseburt) [reply g2core/stepper.cpp]: Well thanks for this discussion, I realized I was doing it wrong as well. I have just pushed a new change to fix where the enable pin settings are stored. Basically, it should be in the StepDirEnable (or whatever the subclasses of Stepper are that care), and be set via a `virtual` function on Stepper that can be overridden in subclasses.

**Pull request merged**